### ISSUE TITLE
Update action to use `node20`

### ISSUE THREAD

**Issue by MikeMcC399** [google-github-actions/release-please-action#852]
### TL;DR

This is an enhancement request to provide a new version of `google-github-actions/release-please-action` which runs under `node20` instead of under `node16`.

### Detailed design

As a minimum the following needs to be updated from node16 to node20:

https://github.com/google-github-actions/release-please-action/blob/910801217bf1ec07dd4367e66345f83918f7ae57/action.yml#L170-L172

### Additional information

The current release [google-github-actions/release-please-action@v3.7.12](https://github.com/google-github-actions/release-please-action/releases/tag/v3.7.12) runs under `node16`.

The blog post [GitHub Actions: Transitioning from Node 16 to Node 20](https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/) Sep 22, 2023 announces the deprecation process in GitHub Actions. Node.js `16` itself already entered [end-of-life](https://nodejs.org/en/blog/announcements/nodejs16-eol) on Sep 11, 2023.

**Comment by MikeMcC399** [comment]
It would also be advisable to update internally-used GitHub Action [workflows](https://github.com/google-github-actions/release-please-action/tree/main/.github/workflows) to use `node20`-compatible actions.

**Comment by MikeMcC399** [comment]
@chingor13 

- I see that you completed PR #863 :+1

Do you plan a new release soon?

**Comment by MikeMcC399** [comment]
[v4.0.0](https://github.com/google-github-actions/release-please-action/releases/tag/v4.0.0) includes support for `node20`.
